### PR TITLE
 all-tests.sh

### PR DESCRIPTION
**Pull request by Innovatorcloudy**
Added a log() function for timestamped messages
Improved error handling

### CODE REVIEW

File: scripts/all-tests.sh [hunk 1]
```diff
@@ -1,34 +1,48 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+log() {
```
Comment (kardolus) [inline scripts/all-tests.sh:4]: **praise**: Nice!

File: scripts/all-tests.sh [hunk 2]
```diff
@@ -1,34 +1,48 @@
...
 
+# Run tests in parallel for faster execution
+log "Running unit tests..."
 cd "$( dirname "${BASH_SOURCE[0]}" )/.."
-./scripts/unit.sh
-echo
-./scripts/integration.sh
-echo
-./scripts/contract.sh
+./scripts/unit.sh &
```
Comment (kardolus) [inline scripts/all-tests.sh:36]: **issue:** I tested this out just now (note that it is still early :)): 
```
./scripts/unit.sh & unit_pid=$!; ./scripts/integration.sh & integration_pid=$!; wait $unit_pid $integration_pid
```
I made my unit test fail, but the integration tests still ran and the exit code was 0:
```
± |main {6} U:1 ✗| → echo $?
0
```
We would want to fail with a non zero exit code if any of the steps fail (including the test). I think we could skip all further steps if one step fails if that makes sense. Ie. don't run integration tests when unit tests fail. Don't run unit tests when the linter fails etc. We could print a message in which we explain that steps get skipped though? What do you think?

**Review by kardolus — COMMENTED** [review]
This is great! Much better than what we had. I think we are almost ready to merge it in.

File: scripts/all-tests.sh (hunk 2 above)
Comment (Innovatorcloudy) [reply scripts/all-tests.sh:36]: Nice thought @kardolus

Comment (Innovatorcloudy) [reply scripts/all-tests.sh:36]: @kardolus Sir will you  merge it?

Comment (kardolus) [reply scripts/all-tests.sh:36]: @Innovatorcloudy I can merge it in and then make a few changes myself. Or you can do them if you'd prefer. Either way works for me.

**Pull request merged**